ingenialink/servo.py

```diff
@@ -1283,8 +1287,43 @@ def read(
         if _reg.dtype == REG_DTYPE.BYTE_ARRAY_512:
             return raw_read
         value = convert_bytes_to_dtype(raw_read, _reg.dtype)
+        self._notify_register_update(_reg, value)
         return value
 
+    def register_update_subscribe(self, callback: Callable[["Servo", Register], None]) -> None:
+        """Subscribe to register updates.
+        The callback will be called when a read/write operation occurs.
+
+        Args:
+            callback: Callable that takes a Servo and a Register instance as arguments.
+
+        """
+        self.__register_update_observers.append(callback)
+
+    def register_update_unsubscribe(self, callback: Callable[["Servo", Register], None]) -> None:
+        """Unsubscribe to register updates.
+
+        Args:
+            callback: Subscribed callback.
+
+        """
+        self.__register_update_observers.remove(callback)
+
+    def _notify_register_update(self, reg: Register, data: Union[int, float, str]) -> None:
+        """Notify a register update to the observers.
+        The updated value is stored in the register's storage attribute.
+
+        Args:
+            reg: Updated register.
+            data: Updated value.
+
+        """
+        for callback in self.__register_update_observers:
+            callback(
+                self,
+                Register(reg.dtype, reg.access, reg.identifier, subnode=reg.subnode, storage=data),
```

Comment:
The register object can hold multiple information (UID, subnode, value) in a single object.

Reply:
I don't understand, creating a new register we lose info, such as register type, enums, etc... 
In addition, should we also send it if the notification comes from a read or a write?

Reply:
I did it this way to avoid modifying the servo's register objects. But now I think there should be no problems since all registers will be reread before storing a configuration file.